--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -1,5 +1,12 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_library")

+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+cc_library(
+    name = "glfw_srcs",
+    textual_hdrs = glob(["glfw/src/**/*.c", "glfw/src/**/*.h", "glfw/include/**/*.h", "glfw/deps/**/*.h", "glfw/deps/**/*.c"]),
+)
+
 go_library(
     name = "glfw",
     srcs = [
@@ -27,6 +34,7 @@
         "window.c",
         "window.go",
     ],
+    cdeps = [":glfw_srcs"],
     cgo = True,
     clinkopts = select({
         "@io_bazel_rules_go//go/platform:android": [
